content/minfraud/normalizing-email-addresses-for-minfraud.mdx

```diff
@@ -17,21 +17,32 @@ you enable sending the MD5 hash. This is the recommended way to do this.
 If you are not able to use our API clients, you can normalize an email address
 yourself. Below are the steps to take to do this.
 
-Trim whitespace from both ends of the address
-
+1. Trim whitespace from both ends of the address.
 1. Lowercase the address.
 1. Find the local part of the email (before the `@`) and the domain (after the
 `@`).
-1. Trim whitespace from both ends of the domain.
-1. If the domain ends with a period, trim it off.
+1. Trim whitespace from the beginning of the domain.
+1. If the domain ends with any number of periods, trim them off.
+1. Check for typos in the domain names (for a complete list of typos we correct
+for, you can consult the [normalization code in one of our client
+APIs](#examples) below).
+1. If the domain ends with a repetition of `.com` (`.com.com`, `.com.com.com`,
+etc.), replace with a single `.com`.
+1. If the domain ends with `.com` followed by any extra characters, the extra
+characters are removed (i.e., `.comxyz` is replaced with `.com`).
+1. If the domain is `gmail.com` with any leading digits, it is replaced with
+`gmail.com` (i.e., `123gmail.com` is replaced with `gmail.com`).
+1. If the domain is `fastmail.com`, remove subdomains (i.e., `user.fastmail.com`
+is replaced with `fastmail.com`).
 1. Convert international domain names (IDNs) to ASCII. For example, you can do
 this in Java using [java.net.IDN](https://docs.oracle.com/javase/8/docs/api/java/net/IDN.html).
```

Comment:
It might be good to move the IDN conversion step earlier - it could go immediately after the trimming periods step - since that's more accurate to what our code does.